invenio_pidstore/providers/base32.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2018 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Douglas Crockford Base32-URL encoder.
+
+This encoder/decoder:
+- uses Douglas Crockford Base32 encoding
+- allows for checksum
+- encodes the checksum using only characters in the base32 set
+  (only digits in fact)
+- produces string that are URI-friendly (no '=' or '/' for instance)
+
+This is based on:
+- https://github.com/datacite/base32-url
+- https://github.com/jbittel/base32-crockford
+"""
+import string
+
+import six
+
+# NO i, l, o or u
+ENCODING_CHARS = '0123456789abcdefghjkmnpqrstvwxyz'
+DECODING_CHARS = {c: i for i, c in enumerate(ENCODING_CHARS)}
+
+
+def encode(number, split_every=0, checksum=False):
```

Comment:
**Moderate**: [`range`default for `split_every` is 0](https://docs.python.org/3/library/stdtypes.html#range). However, if you explicitly pass `0`, it fails, so default value would never work.

``` python
>>> range(1, 10, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: range() arg 3 must not be zero
```

Reply:
Splitting seemed like a bonus, so I didn't put it as a default. Also, at this point, there is no foreknowledge of the final string length: 4 might be a poor choice for the passed number causing an unbalanced split... If people feel strongly about it I can change it.

Reply:
@ppanero since we are checking for that case on line 63, we should be fine right?

```python
if split_every:  # False for 0 so range(..., ..., 0) never occurs. 
```

Reply:
True that, plus if it's gonna be a "full generic rand string generator" it should not imply any split by default.